contents/convolutions/code/csharp/1DConvolution.cs

```diff
@@ -0,0 +1,109 @@
+﻿using System;
+using System.IO;
+
+namespace Convolution1D
+{
+    public class Convolution1D
+    {
+        // Creates a sawtooth function with the given length
+        static double[] CreateSawtooth(int length)
+        {
+            var array = new double[length];
+            for (var i = 0; i < length; i++)
+                array[i] = (i + 1) / 200f;
+            return array;
+        }
+
+        // Normalizes the given array
+        static void Normalize(double[] array)
+        {
+            var norm = Norm(array);
+            for (var i = 0; i < array.Length; i++)
+                array[i] /= norm;
+        }
+
+        // Calculates the norm of the array
+        static double Norm(double[] array)
+        {
+            var sum = 0.0;
+            for (var i = 0; i < array.Length; i++)
+                sum += Math.Pow(array[i], 2);
+            return Math.Sqrt(sum);
+        }
+
+        // Julia like modulus function which handles negative values properly
+        // Assumes that y >= 0
```

Comment:
The [styleguide](https://docs.microsoft.com/en-us/dotnet/csharp/fundamentals/coding-style/coding-conventions) recommends ending your sentences with a '.' 

```suggestion
        // Assumes that y >= 0.
```